### ISSUE TITLE
String variables with spaces 

### ISSUE THREAD

**Issue by raymondfeng** [jsonata-js/jsonata#528]
With the following json doc:

```json
{
  "countByTeam": {
    "Team 1": 5,
    "Team 2": 1
  }
}
```

I would like to find if `Team 1` has a value >= `<minCount>` using `$[countByTeam.$teamName >= $minCount]`.

It fails with bindings:

```js
{
 teamName: "Team 1",
 minCount: 1
}
```
https://try.jsonata.org/Gflyus8B5

I also tried to escape the team name with `. and it also failed.

```
$[countByTeam.`$teamName` >= $minCount]
```

**Comment by markmelville** [comment]
use the `$lookup` function:
```
$[$lookup(countByTeam,$teamName) >= $minCount]
```

**Comment by raymondfeng** [comment]
@markmelville Thank you for the quick response. `$lookup` works!

**Comment by markmelville** [comment]
> Do you have plan to allow the syntax I use?

I'm not a maintainer of this repo, nor do I have any input into the roadmap. I think it would be cool if there was a syntax to do it. I think your guess was that the backticks are a [string interpolation](#256) syntax, but they're not. It's an escape sequence. I might claim that it's not needed as an escape sequence, since quotes can escape just fine, but I'm sure someone who wrote JSONata would be able to say why backticks are yet another escape sequence.

**Comment by andrew-coleman** [comment]
>  I might claim that it's not needed as an escape sequence, since quotes can escape just fine

That's true for examples like `object[countByTeam.'$teamName' > 1]` because the parser recognizes that the 'string' token cannot be a string in that context and changes it to a 'name' token.  However, if the expression was something like `object.countByTeam['$teamName' > 1]`, then the parser will just assume that `'$teamName'` is a string literal.  Originally, the workaround for this situation was to use `object.countByTeam[$.'$teamName' > 1]` to put that string token in a path context, but this was not obvious to users.  So I added the backticks to indicate that the string should be treated as a name token in any context and I intended to deprecate the use single or double quotes for this purpose (although I've never done anything about that).

BTW, I like the idea of a string interpolation syntax.  I just need to figure out what the syntax should be (ideas always welcome).

Thanks :)
